### ISSUE TITLE
Messages not showing on new conversations

### ISSUE THREAD

**Issue by HypoLuxa** [snikket-im/snikket-ios#194]
**Describe the bug**
Message not showing when there are no previous messages from sender, though there is a notification of a message

**To Reproduce**
1. Snikket installed on iOS
2. Notifications enabled (I only have badges enabled)
3. Receive a message from a new sender who you have not received from before (no previous message from sender in 'chats' tab)
4. Badge shows new message
5. There is no new message in 'chats' tab
6. Badge gets stuck showing new notification

**Screenshots**
Unfortunately nothing to show since there is no message visible

**Software versions:**
 - App Version: 1.2
 - iOS version: 15.4.1
 - iPhone model: iPhone 13 pro max

 - Server software: jmp.chat

If your problem involves communication with people using other apps/platforms,
please include details of their app and versions also when possible.

**Additional context**
As far as i know this happens in the chats tab, I have not received voice calls, it may happen, but I am not aware. This is not from other xmpp users that I am aware, this would be from conventional sms numbers.

To fix, I had to use a different client to see who the message was from and add that contact to my contacts and start a chat with that user and the message would show and badge would disappear

Thank you, am willing to troubleshoot.

**Comment by mwild1** [comment]
Thanks for the details! Is your account on a Snikket server, or a different (non-Snikket) XMPP service?

**Comment by HypoLuxa** [comment]
on a service, I think I mentioned in the issue its jmp.chat

**Comment by glward** [comment]
I’ve seen this as well, as have a few folks with Siskin: tigase/siskin-im#146 

In my case I’m using Snikket as my XMPP host, but I’m also using JMP.chat. I suspect this is going to be mostly reported by people using JMP.chat since “receiving messages from people not in your roster” is pretty uncommon, but “getting SMS messages from random numbers” is quite common.

I’ve seen this with iOS and iPadOS, running the latest Snikket server and app (1.2). 

Other apps (Monal on iOS/iPadOS/macOS, Cheogram and Conversations on Android, Dino on Linux) have no issues here — it’s something with Siskin/Snikket iOS. 

I have seen it work if I get a message _while the app is open_, but even that is not consistent.

**Comment by fjrider** [comment]
I am also seeing this issue, following and hoping for a resolution:

Snikket Version 1.2, Build: 10
iOS Version: 16.4.1 (20E252)
Using a snikket.chat account, but with the jpm.chat service.

Willing to try things out with Snikket team to hopefully find a workaround or fix soon.

**Comment by mwild1** [comment]
Thanks for the additional information!

I have a potential fix locally that I'm aiming to publish to the beta builds soon for testing. I'll let you know when it's available (unfortunately I need to replace my Mac before I can upload any more builds :slightly_frowning_face: ).

**Comment by fjrider** [comment]
Any update on this?

Any way I could help tinker or test?

**Comment by ajwhouse** [comment]
Hi, I am having this problem and it is the main thing preventing me from adopting JMP.chat w/ Snikket. Happy to help resolve it if I can be of use. @mwild1 you mention a potential local fix. Anything come of that?

**Comment by HypoLuxa** [comment]
Yeah I’m the one who opened this 18 months ago, because of this bug I’ve abandoned jump.chat/xmpp totally. Unfortunate.

**Comment by mwild1** [comment]
I'm curious if anyone is still experiencing this in the latest app store release (1.2.1) with the latest Snikket server (February). Both included fixes for bugs that could have contributed to the behaviour described by reporters here.

**Comment by glward** [comment]
Just did a quick test, with 1.2.1 build 12 (TestFlight)... and discovered I never updated the docker tag from beta -> stable on my server, whoops, but it let me test:

Server stable.20240115: did not work
Server stable.20240221: did work

So maybe it's fixed?

**Comment by glward** [comment]
Nope, still not fixed. Just got a message notification but no message shows up in the chats list, same behavior as before.

**Comment by ERamseth** [comment]
I found this issue by an internet search for the same behavior. 

On snikket 1.2.1 build 12 on iOS and behavior is as the OP described. Server is snikket hosted service.

**Comment by markginter31** [comment]
I have two people on iPhones who see this.  I'm running a small snikket server for a small group of people.  Things were working fine until I moved the server instance off of a vm and onto a physical machine.  Is it possibly related to renewing the SSL certificate?  Maybe something changed with OMEMO keys?  Is there a way to reset the certificate on the iPhone client side?  Really spitting in the wind on this one

**Comment by mwild1** [comment]
I don't have any evidence that it is related to SSL certificates or OMEMO (obviously unless we figure out what it *is* then it's pretty hard to rule out anything as a potential factor).

If this is a persistent problem for you, you can explore a couple of other options:

- [Monal](https://monal-im.org/) is an alternative iOS app that is mostly compatible with Snikket (we're just hoping they will work on improving compatibility, e.g. https://github.com/snikket-im/snikket-server/issues/240 can cause issues). You can just put your Snikket credentials into Monal and it should just work.
- If you're feeling significantly adventurous, we are working on a new iOS app for Snikket, and the first alpha was made available for testing this week as "Cheogram iOS". It's at a young seedling stage right now - in particular OMEMO support and video calls are notably lacking, and it *definitely* has bugs. Also, some features (including notifications) require switching to the development branch of snikket-server.
Until it gets a bit more polish it won't be available under the "Snikket" brand for now. But hopefully that's not too far off. Anyone who is interested can participate in testing via this [TestFlight link](https://testflight.apple.com/join/rmfXPT3S).

**Comment by HypoLuxa** [comment]
“I cant implement or fix a basic feature, use something else” GeezOn Jul 25, 2025, at 6:00 AM, Matthew Wild ***@***.***> wrote:﻿mwild1 left a comment (snikket-im/snikket-ios#194)
I don't have any evidence that it is related to SSL certificates or OMEMO (obviously unless we figure out what it is then it's pretty hard to rule out anything as a potential factor).
If this is a persistent problem for you, you can explore a couple of other options:

Monal is an alternative iOS app that is mostly compatible with Snikket (we're just hoping they will work on improving compatibility, e.g. snikket-im/snikket-server#240 can cause issues). You can just put your Snikket credentials into Monal and it should just work.
If you're feeling significantly adventurous, we are working on a new iOS app for Snikket, and the first alpha was made available for testing this week as "Cheogram iOS". It's at a young seedling stage right now - in particular OMEMO support and video calls are notably lacking, and it definitely has bugs. Also, some features (including notifications) require switching to the development branch of snikket-server.
Until it gets a bit more polish it won't be available under the "Snikket" brand for now. But hopefully that's not too far off. Anyone who is interested can participate in testing via this TestFlight link.

**Comment by mwild1** [comment]
> “I cant implement or fix a basic feature, use something else” Geez

We're a tiny dev team, and yes, this means we can't do everything we want to do. If you're able to fix this issue/"basic feature", then your contributions are welcome. If not, and you don't like the options I suggested, you're certainly more than welcome to use something else. Posting non-constructive comments just wastes your time and ours.

We specifically chose [not to seek investment](https://snikket.org/blog/snikket-cic/) from VCs or start our own cryptocurrency. Even though those would have been a shortcut to having more dev resources available, they don't align with our values. Many other alternatives to Snikket are available which *did* follow these business models, and you're welcome to use them.

Our chosen path is certainly a little slower, but we're doing our best to build a long-term healthy project 🙂

**Comment by P-L-K** [comment]
I’m here to confirm this issue still persists on jmp.chat and snikket. I have been using movim on safari to add the sender of the conversation to my contacts, then it shows up on snikket on future interactions.
